### ISSUE TITLE
method connect hangs until driver is used in another thread

### ISSUE THREAD

**Issue by dbacvetkov** [oracle/python-oracledb#408]
Hello!

My application starts multiple threads and every thread is trying to connect to different oracle database, using this module (2.4.1) in thin mode.
So every thread is creating his own independent connection.
It turned out that method connect hangs until driver is used in another thread.
To be more specific - until driver is creating connection in another thread, which may take a while bc of network or load issues.

I have made research and found out that it hangs in this place in command "manager.condition.wait()"

But funny part is if i comment out this string ( manager.condition.wait() ) - it works perfectly fast and well ! :) 

So... Why? :)

```
def get_manager(requested_thin_mode=None):
    """
    Returns the manager, but only after ensuring that no other threads are
    attempting to initialize the mode.

    NOTE: the current implementation of the driver only requires
    requested_thin_mode to be set when initializing the thick mode; for this
    reason the error raised is specified about a thin mode connection already
    being created. If this assumption changes, a new error message will be
    required.
    """
    with manager.condition:
        if manager.thin_mode is None:
            if manager.requested_thin_mode is not None:
                manager.condition.wait()      # this place
            if manager.thin_mode is None:
                if requested_thin_mode is None:
                    manager.requested_thin_mode = True
                else:
                    manager.requested_thin_mode = requested_thin_mode
        elif (
            requested_thin_mode is not None
            and requested_thin_mode != manager.thin_mode
        ):
            errors._raise_err(errors.ERR_THIN_CONNECTION_ALREADY_CREATED)
    return manager
```

**Comment by anthony-tuininga** [comment]
This code was written to have the following behavior:

- only thin mode or only thick mode can be enabled at any one time
- the mode is set *after* the first connection is created (thin mode) or thick mode is enabled
- if a connection fails to be created in thin mode, an attempt to enable thick mode will still work

Because of this we wait until the first connection has been created before allowing subsequent connections to be created. In order to help your situation, however, we could create a new method:

```python
def enable_thin_mode():
    """
    Enables thin mode only. After this method is called enable_thick_mode() will fail. Prevents the
    requirement of establishing a connection first before allowing multiple threads to run.
    """
```

Thoughts?

**Comment by dbacvetkov** [comment]
@anthony-tuininga 
Yes, i think it would be great!
Thank you :) 

PS. My app is a service of availability checking of thousands of different databases, some of which is unavailable sometime. 
This check happens twice a minute )) 
That`s why it so important to check (and give a timeout error) in a few seconds )

**Comment by cjbj** [comment]
Separate from the mode determination issue, would it be nicer to the DBs to keep connections open, and then just verify the DB availability with a `connection.ping()` call?  Yes this would retain some memory on the DB host when you're not pinging, but it would stop the server having to create & destroy a server process each time you create a connection when you want to check availability.  Or you could go half-way and use a connection pool with min=0 and some decent max_lifetime_session value so that if you don't ping for a while, then the server process will go away.

**Comment by dbacvetkov** [comment]
@cjbj thank you for your idea, and i agree with you that this makes sense and saves servers resources.
But i have a task to check ability to connect to the database, which in case of already existing connection is something different.
Imaging, your database reached sessions limit, and your clients (your enterprise applications) unable to connect. But availability checking service is using existing connection and reports that all is OK :)

**Comment by cjbj** [comment]
@dbacvetkov I get it.  (But imagine that your checking program uses the last session before the limit it reached !)

**Comment by cjbj** [comment]
@anthony-tuininga based on your internal comment about zero size pools enabling Thin mode, I imagine the following line is a valid workaround for @dbacvetkov:

```
oracledb.create_pool(user="x", password="y", dsn="localhost/dummy", min=0, max=1)
```

**Comment by anthony-tuininga** [comment]
Yes, that works as a workaround.

**Comment by cjbj** [comment]
@dbacvetkov How are you actually reporting/monitoring the stats?  We're interested to hear from people using OpenTelemetry e.g. via the opentelemetry-instrumentation-dbapi module.

**Comment by dbacvetkov** [comment]
@cjbj I don't use opentelemetry.
My service publishes results via prometheus exporter (module prometheus_client).

**Comment by anthony-tuininga** [comment]
I have pushed a patch that adds have requested support and have initated a [build](https://github.com/oracle/python-oracledb/actions/runs/11468908376) from which you can download pre-built development wheels once it completes. You can also [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) if you prefer.

**Comment by anthony-tuininga** [comment]
This was included in python-oracledb 2.5.0 which was just released.
